### PR TITLE
Reactive tools search with JS

### PR DESCRIPTION
**Pull request by domasV**

### CODE REVIEW

File: src/Controller/ApiController.php [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    }
+    /**
+     * @Route("/", name="api_index")
+     */
+    public function index(): Response
+    {
+        return $this->render('api/index.html.twig', [
+            'controller_name' => 'ApiController',
+        ]);
+    }
```
Comment (aivarasbaranauskas) [inline src/Controller/ApiController.php:32]: Unused method, should be removed.

**Review by aivarasbaranauskas — CHANGES_REQUESTED** [review]

File: src/Controller/ApiController.php (hunk 1 above)
Comment (domasV) [reply src/Controller/ApiController.php]: Removed

File: public/js/tool-search.js [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+            element.addEventListener('click', function() {
+                searchString.value = this.getAttribute('data-tag')
+                searchTools();
+            })
+        }
+        clear_search.addEventListener('click', function() {
+            searchString.value = ""
+            searchTools();
+        })
+        searchString.addEventListener('keyup', searchTools)
```
[the remark this answers is not in the document]
Comment (domasV) [reply public/js/tool-search.js:28]: Added delay with variable. Variable set for 1 second, which feels natural, might need to fine-tune according more opinions :D

[the remark this answers is not in the document]
Comment (aivarasbaranauskas) [reply public/js/tool-search.js:28]: Debounce feel buggy, if I type something in a 1s window after last time search was done it does not trigger new search later.

I would rather have used setTimeout() than calculating intervals manually. A good example: https://www.freecodecamp.org/news/javascript-debounce-example

**Review by aivarasbaranauskas — CHANGES_REQUESTED** [review]

File: public/js/tool-search.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (domasV) [reply public/js/tool-search.js:27]: Agree. Changed to the provided debounce algorithm.

**Review by aivarasbaranauskas — APPROVED** [review]

**Pull request merged**